### DISCUSSION TITLE
Why is a highly uniquely expressed gene not being detected as significant marker?

### DISCUSSION THREAD

**Discussion by denvercal1234GitHub** [satijalab/seurat#4706] [Q&A] [answered]
Hi there, 

When I plotted a gene on RNA Assay, it is evident that this gene is highly expressed, but it was not being identified as a significant marker for that cluster using wilcox test in FindAllMarkers(). It is also not present in the output list of all (significant and non-significant) markers, so it is unlikely the issue due to strictness of p-value correction method. 

Does it mean the choice of wilcox test for DEG of my data (SCTransformed-ed) is not ideal? Or, is it some bug with the function?

Thank you for your help!

<img width="610" alt="Screen Shot 2021-06-30 at 12 50 00 AM" src="https://user-images.githubusercontent.com/62179082/123881665-1b594600-d93d-11eb-94f0-37b03e94b077.png">

**Comment by saketkc** [comment]
Can you paste the code you use to find the markers?

**Reply by denvercal1234GitHub** [reply]
Hi there, 

Below are the steps before FindAllMarkers() code. Thank you for your help!

1.	Create all Seurat objects
2.	QC and filter each object 
3.	Calculate percent.mt on raw counts for each object
4.	Merge all objects
5.	**Normalize by SCTransform(): scale = FALSE, center = FALSE, no regression of any variable**
6.	Calculate cell cycle scores on SCT-normalized data
7.	Find cc score difference on SCT-normalized data
8.	**Switch to RNA assay** 
9.	**SCTransform(scale = TRUE, center = TRUE) and regressing percent.mt and cc.difference** 
10.	**Switch to RNA Assay** 
11.	**Normalize each object by NormalizeData(assay = RNA)** 
12.	**ScaleData(assay = RNA)** 
13.	Switch to SCT Assay to perform PCA and clustering
14.	RunPCA() with features = VariableFeatures() [on SCT Assay]
15.	FindNeighbors()
16.	FindCluster(Louvain, resolution = 1) [on SCT Assay]
17.	RunUMAP() [on SCT Assay] 
18.	Perform RNA trajectory, cell type annotation analyses [on SCT Assay]

```
DefaultAssay (object) <- 'RNA'

object <- FindAllMarkers(object, only.pos = F, assay = "RNA")

```

**Reply by saketkc** [reply]
It might be easier to for me to comment if I can take a look at the object. You can anonymize the genes if you wish and take a subsample of the cells (where you can replicate this issue). My email is schoudhary@nygenome.org

**Reply by denvercal1234GitHub** [reply]
Thank you very much, @saketkc. I emailed you the object with the title "Seurat Discussion Issue #4706"

**Comment by denvercal1234GitHub** [comment, marked as answer]
Answered via email: Artifact by visualization of dotplot.
